test/HttpRequestTests.cpp

```diff
@@ -0,0 +1,371 @@
+    int curStatusCodeClass = -1;
+    for (int statusCode = 100; statusCode < 600; ++statusCode)
+    {
+        const std::string url = "/status/" + std::to_string(statusCode);
+        httpRequest.setUrl(url);
+
+        timedout = true; // Assume we timed out until we prove otherwise.
+
+        httpSession->asyncRequest(httpRequest, pollThread);
+
+        std::unique_lock<std::mutex> lock(mutex);
+        cv.wait_for(lock, std::chrono::seconds(1));
+
+        const std::shared_ptr<const http::Response> httpResponse = httpSession->response();
+        LOK_ASSERT(httpResponse->state() == http::Response::State::Complete);
+        LOK_ASSERT(!httpResponse->statusLine().httpVersion().empty());
+        LOK_ASSERT(!httpResponse->statusLine().reasonPhrase().empty());
+
+        if (statusCode % 100 == 0)
+            ++curStatusCodeClass;
+        LOK_ASSERT(httpResponse->statusLine().statusCategory()
+                   == statusCodeClasses[curStatusCodeClass]);
+
+        LOK_ASSERT(httpResponse->statusLine().statusCode() == statusCode);
+
+        if (httpResponse->statusLine().statusCategory()
+            != http::StatusLine::StatusCodeClass::Informational)
+        {
+            const auto pocoResponse = pocoGet(host, url); // Get via Poco in parallel.
+            compare(*pocoResponse.first, pocoResponse.second, *httpResponse);
+        }
+    }
+
+    pollThread.joinThread();
+}
+
+void HttpRequestTests::testSimplePost()
+{
+    const std::string Host = "httpbin.org";
+    const char* URL = "/post";
+
+    // Start the polling thread.
+    SocketPoll pollThread("HttpSessionPoll");
+    pollThread.startThread();
+
+    http::Request httpRequest(URL, http::Request::VERB_POST);
+
+    // Write the test data to file.
+    const char data[] = "abcd-qwerty!!!";
+    const std::string path = FileUtil::getSysTempDirectoryPath() + "/test_http_post";
+    std::ofstream ofs(path, std::ios::binary);
+    ofs.write(data, sizeof(data) - 1); // Don't write the terminating null.
+    ofs.close();
+
+    httpRequest.setBodyFile(path);
+
+    auto httpSession = http::Session::createHttp(Host);
+    httpSession->setTimeout(std::chrono::seconds(1));
+
+    std::condition_variable cv;
+    std::mutex mutex;
+    bool timedout = true;
+    httpSession->setFinishedHandler([&](const std::shared_ptr<http::Session>&) {
+        std::lock_guard<std::mutex> lock(mutex);
+        timedout = false;
+        cv.notify_all();
+    });
+
+    httpSession->asyncRequest(httpRequest, pollThread);
+
+    std::unique_lock<std::mutex> lock(mutex);
+    cv.wait_for(lock, std::chrono::seconds(1));
+
+    const std::shared_ptr<const http::Response> httpResponse = httpSession->response();
+    LOK_ASSERT(httpResponse->state() == http::Response::State::Complete);
+    LOK_ASSERT(!httpResponse->statusLine().httpVersion().empty());
+    LOK_ASSERT(!httpResponse->statusLine().reasonPhrase().empty());
+    LOK_ASSERT(httpResponse->statusLine().statusCode() == 200);
```

Comment:
Hmm, this worked yesterday, but now it returns 403. Do you think you could hide these tests (that rely on httpbin.org accepting our requests) behind some flag, so tests don't start failing just because an external service is down? At least both me and @tml1024 get failures now. Thanks.